core/src/storage/collections/binary_heap/impls.rs

```diff
@@ -440,3 +440,29 @@ where
         Iter::new(self)
     }
 }
+
+impl<T> Extend<T> for BinaryHeap<T>
+where
+    T: Codec + Ord,
+{
+    fn extend<I>(&mut self, iter: I)
+    where
+        I: IntoIterator<Item = T>,
+    {
+        for i in iter {
+            self.push(i)
+        }
+    }
+}
+
+impl<'a, T: 'a + Copy> Extend<&'a T> for BinaryHeap<T>
+where
+    T: Codec + Ord,
+{
+    fn extend<I>(&mut self, iter: I)
+    where
+        I: IntoIterator<Item = &'a T>,
+    {
+        self.extend(iter.into_iter().cloned())
+    }
+}
```

Comment:
Is this standard Rust API design for `Extend` implementations? (real question)
Also why not use `Iterator::copied` instead of `Iterator::cloned` if `T: Copy`?
Also please put all trait bounds into the where clause.

Reply:
Same question as above applies to the other data structures and their `Extend` implementation.

Reply:
Yeah I copied this pattern from the standard library. e.g: 
- https://doc.rust-lang.org/src/alloc/collections/btree/map.rs.html#1879
- https://doc.rust-lang.org/src/std/collections/hash/map.rs.html#2427

Reply:
But I didn't know about `Iterator::copied`: that would make more sense. Maybe an opportunity for a nice and easy PR to rust std lib :)

Reply:
Okay if copied from standard library I assume it makes sense to have to keep ink! more in sync with expections from the standard libs.